src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -2058,13 +2061,18 @@ public int hashCode() {
             if (firstOnly) return ret;
         }
 
-        if (this.getAlignmentStart() != NO_ALIGNMENT_START && this.getIndexingBin() != null &&
-                this.computeIndexingBin() != this.getIndexingBin()) {
-            if (ret == null) ret = new ArrayList<>();
-            ret.add(new SAMValidationError(SAMValidationError.Type.INVALID_INDEXING_BIN,
-                    "bin field of BAM record does not equal value computed based on alignment start and end, and length of sequence to which read is aligned",
-                    getReadName()));
-            if (firstOnly) return ret;
+        if (this.getAlignmentStart() != NO_ALIGNMENT_START) {
+            if (getHeader() != null
+                    && isReferenceTooLargeForBAI()
+                    && getValidationStringency() != ValidationStringency.SILENT) {
+                LOG.warn(String.format("Reference length is too large for BAM bin field for %s record. Values in the bin field could be incorrect.", getReadName()));
+            } else if (isIndexingBinNotEqualsComputedBin()) {
+                if (ret == null) ret = new ArrayList<SAMValidationError>();
```

Comment:
According to codecov this code is not covered by a test, it should be.

Reply:
Done.